From bfc6d25318e3bbf3d9c0389b33ad45ae1bc401f0 Mon Sep 17 00:00:00 2001
From: Daniel Bevenius <daniel.bevenius@gmail.com>
Date: Tue, 28 Feb 2017 20:04:12 +0100
Subject: [PATCH] src: add HAVE_OPENSSL directive to openssl_config

Currently when building with the following configuration options:
$ ./configure --without-ssl && make

The following link error is reported:

Undefined symbols for architecture x86_64:
  "node::openssl_config", referenced from:
      node::Init(int*, char const**, int*, char const***) in node.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see
invocation)

Adding an HAVE_OPENSSL directive around this code allows the build to
pass.

PR-URL: https://github.com/nodejs/node/pull/11618
Reviewed-By: Anna Henningsen <anna@addaleax.net>
Reviewed-By: James M Snell <jasnell@gmail.com>
Reviewed-By: Colin Ihrig <cjihrig@gmail.com>
Reviewed-By: Ben Noordhuis <info@bnoordhuis.nl>
Reviewed-By: Sam Roberts <vieuxtech@gmail.com>
[Martin: patch from https://github.com/nodejs/node/pull/11734]
Signed-off-by: Martin Bark <martin@barkynet.com>
---
 src/node.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/node.cc b/src/node.cc
index 35d29bd..68362a7 100644
--- a/src/node.cc
+++ b/src/node.cc
@@ -4203,8 +4203,10 @@ void Init(int* argc,
         SafeGetenv("NODE_PRESERVE_SYMLINKS", &text) && text[0] == '1';
   }
 
+#if HAVE_OPENSSL
   if (openssl_config.empty())
     SafeGetenv("OPENSSL_CONF", &openssl_config);
+#endif
 
   // Parse a few arguments which are specific to Node.
   int v8_argc;
